bark/world/evaluation/rss/rss_interface.hpp

```diff
@@ -0,0 +1,249 @@
+using ::ad::physics::Speed;
+
+struct AgentState {
+  float timestamp;
+  Speed speed;
+  Distance max_stopping_distance;
+  ::ad::map::point::ENUPoint center;
+  ::ad::map::point::ENUHeading heading;
+};
+
+namespace bg = boost::geometry;
+namespace bark {
+namespace world {
+
+using geometry::Line;
+using geometry::Point2d;
+using geometry::Polygon;
+using models::dynamic::Trajectory;
+using models::dynamic::StateDefinition::THETA_POSITION;
+using models::dynamic::StateDefinition::TIME_POSITION;
+using models::dynamic::StateDefinition::VEL_POSITION;
+using models::dynamic::StateDefinition::X_POSITION;
+using models::dynamic::StateDefinition::Y_POSITION;
+using objects::AgentPtr;
+
+namespace evaluation {
+
+typedef std::unordered_map<objects::AgentId, bool> PairwiseEvaluationReturn;
+typedef std::unordered_map<objects::AgentId, std::pair<bool, bool>>
+    PairwiseDirectionalEvaluationReturn;
+
+// Generates a RSS world model from information of agents from BARK, which
+// serves as the input of RSS check. Performs RSS check and returns safety
+// responses the specifed agent.
+//
+// Mostly follows the implements of Carla-RSS integration
+//
+// Using libraries:
+// RSS: https://github.com/intel/ad-rss-lib
+// ad_map_access (dependency of RSS): https://github.com/carla-simulator/map
+//
+// Carla-RSS integration:
+// https://github.com/carla-simulator/carla/tree/master/LibCarla/source/carla/rss
+class RssInterface {
+ public:
+  RssInterface() {}
+  explicit RssInterface(const std::string& opendrive_file_name,
+                        const std::vector<float>& default_vehicle_dynamics,
+                        const std::unordered_map<AgentId, std::vector<float>>&
+                            agents_vehicle_dynamics,
+                        const float& checking_relevent_range,
+                        const float& route_predict_range)
+      : default_dynamics_(default_vehicle_dynamics),
+        agents_dynamics_(agents_vehicle_dynamics),
+        checking_relevent_range_(checking_relevent_range),
+        route_predict_range_(route_predict_range) {
+    // Sanity checks
+    assert(opendrive_file_name != "");
+    assert(checking_relevent_range_ >= 1.);
+    // lon_max_brake<=lon_min_brake && lon_min_brake<=lon_min_brake_correct
+    assert(default_dynamics_.size() == 8);
+    assert(default_dynamics_[1] <= default_dynamics_[2] &&
+           default_dynamics_[2] <= default_dynamics_[3]);
+    for (const auto& d : agents_dynamics_) {
+      assert(d.second.size() == 8);
+      assert(d.second[1] <= d.second[2] && d.second[2] <= d.second[3]);
+    }
+
+    // the debugging level in RSS
+    spdlog::set_level(spdlog::level::off);
+    initializeOpenDriveMap(opendrive_file_name);
+    rss_coordinate_transform_.setENUReferencePoint(
+        ::ad::map::access::getENUReferencePoint());
+  }
+
+  // Returns a boolean indicating the safety response of the specified agent.
+  // True if for each nearby agents, at least one of the all possible RSS
```

Comment:
I do not get that comment. Should it not return true if it violates one of the possible RSS situations

Reply:
I think my way to explain it in the comment is somewhat confusing, I will try to write it more detailly later.

And additionally, let me explain the logic of rss check by a drawing
![image](https://user-images.githubusercontent.com/22342030/93650674-7d306b00-fa0f-11ea-825c-2cb2ebfada3c.png)

We have 3 situations here
1. In situation 1, the blue agent is with a safe distance to ego longitudinally (or vertical), so it is longitudinally safe. But if we just look at the latitudinal distance (just horizontal), they don't have a safe distance. I refer each safety situation with a different direction as a safety situation. So if one of the situation is safe, intuitively the ego is overall safe.
2. In 2, with the same logic, it is latitudinally safe but not longitudinally.
3. In 3, it is both longitudinally and latitudinally safe.